R/model.R

```diff
@@ -386,6 +387,22 @@ sleuth_results <- function(obj, test, test_type = 'wt',
       by = 'target_id')
   }
 
+  if ( aggregate_pval ) {
+  	if(is.null(obj$target_mapping)) {
+  			stop('Must provide transcript to gene mapping table in order to aggregate p-values')
+  	}
+    if(obj$gene_mode) {
+    		stop('Aggregation cannot be done from results in gene mode. Must be performed on results in transcript mode.')
+    }
+    # TODO: p-value aggregation here
+    if(any(res$pval < 10^-323, na.rm=TRUE)) {
+    		warning('Extreme p-values around and below 10^-320 will generate 0 pvalues in aggregation')
+    }
+    res <- data.table::as.data.table(res)[, .(num_aggregated_transcripts = length(!is.na(pval)), sum_mean_obs_counts = sum(mean_obs, na.rm=TRUE), pval = as.numeric(lancaster(pval, exp(mean_obs)))), by=eval(obj$gene_column)]
```

Comment:
Changed to data.table implementation. sleuth_results(pval_aggregate = TRUE) now takes ~1.5 seconds instead of 6 seconds with data.frame.   Please check that I've optimized data.table, @warrenmcg , @pimentel. Thanks!

Reply:
Also I'm going to ask this question here:  I noticed that using a newly generated sleuth object, I get much fewer significant genes compared to my older sleuth object (using sleuth about a year or so ago) on the same data set.  I just want to make sure this is correct; i.e. there were changes in sleuth to support this observation.

Reply:
Looks good to me! I would only suggest a formatting change to keep the columns to under 100 characters ([suggested R style tips](http://adv-r.had.co.nz/Style.html); recommends 80, but I usually go with 100 because of some nesting that happens).

line 401 would read like this:
```
res <- data.table::as.data.table(res)[,
         .(num_aggregated_transcripts = length(!is.na(pval)),
           sum_mean_obs_counts = sum(mean_obs, na.rm=TRUE),
           pval = as.numeric(lancaster(pval, exp(mean_obs)))),
         by = eval(obj$gene_column)]
```

The # of spaces may differ between here and your editor, so just align the period with the 1st 't' in data.table, the other columns with `num_aggregated_transcripts`, and the `by` column with the period.

Reply:
# of genes called issue resolved: false alarm